src/main/java/nl/novi/behuiswerkles11/controllers/TelevisionsController.java

```diff
@@ -1,39 +1,49 @@
 package nl.novi.behuiswerkles11.controllers;
 
-import org.springframework.http.HttpStatus;
-import org.springframework.http.HttpStatusCode;
+import nl.novi.behuiswerkles11.models.Television;
+import nl.novi.behuiswerkles11.repositories.TelevisionRepository;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+import java.util.Optional;
+
 @RestController
 @RequestMapping("televisions")
 public class TelevisionsController {
+    private final TelevisionRepository televisionRepository;
+
+    public TelevisionsController(TelevisionRepository televisionRepository) {
+        this.televisionRepository = televisionRepository;
+    }
+
     @GetMapping()
-    public ResponseEntity<String> getTelevisions() {
-//        return ResponseEntity.ok("televisions");
-        return new ResponseEntity<>("all televisions", HttpStatus.OK);
+    public ResponseEntity<List<Television>> getTelevisions() {
+        return ResponseEntity.ok(televisionRepository.findAll());
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<String> getOneTelevision(@PathVariable Integer id) {
-        return ResponseEntity.ok("televisionId: " + id);
+    public ResponseEntity<Optional<Television>> getOneTelevision(@PathVariable Long id) {
+        return ResponseEntity.ok(televisionRepository.findById(id));
     }
 
     // Omdat we hier iets posten in plaats van aanvragen, hebben we geen body in onze return. Vandaar dat er <void> staat na het definieren van de ResponseEntity.
     @PostMapping()
-    public ResponseEntity<Void> addTelevision(@RequestBody String television) {
-        System.out.println(television);
+    public ResponseEntity<Void> addTelevision(@RequestBody Television television) {
+        televisionRepository.save(television);
         return ResponseEntity.created(null).build();
     }
 
     @PutMapping("/{id}")
-    public ResponseEntity<Object> updateTelevision(@PathVariable int id, @RequestBody String television) {
-        return  ResponseEntity.noContent().build();
+    public ResponseEntity<Television> updateTelevision(@PathVariable Long id, @RequestBody Television television) {
+        televisionRepository.saveAndFlush(television);
```

Comment:
Je kunt hier beter "gewoon" `.save(television)` gebruiken. saveAndFlush doet praktisch hetzelfde, maar is aggressiever, dus je moet het alleen gebruiken als je weet wat de nuance is en je er een goede reden voor hebt.